feat: add lookahead

emjshrx opened the pull request:
fixes #20 . Didnt feel any new tests are needed, lmk if otherwise

theanmolsharma reviewed (changes requested):
PR looks solid apart from one minor change. Nice work!!

About the tests, I think this change should be accompanied by some tests. You can directly check in the db whether 10 addresses are being generated on `init()` and then for subsequent addresses make sure `i + 10`th address is being saved in db.

emjshrx commented:
> PR looks solid apart from one minor change. Nice work!!
> 
> About the tests, I think this change should be accompanied by some tests. You can directly check in the db whether 10 addresses are being generated on `init()` and then for subsequent addresses make sure `i + 10`th address is being saved in db.

Addressed comments
